### PR TITLE
Add red encoding for opus only publisher

### PR DESCRIPTION
**Pull request by cnderrauber**
With this PR, livekit can forward red encodings for opus upstream. For red encoding upstream to opus downstream, can detect packet loss and recover it from the red encoding blocks(not implement yet).

### CODE REVIEW

File: pkg/rtc/mediaengine.go [hunk 1]
```diff
@@ -11,6 +10,7 @@ import (
 )
 
 var opusCodecCapability = webrtc.RTPCodecCapability{MimeType: webrtc.MimeTypeOpus, ClockRate: 48000, Channels: 2, SDPFmtpLine: "minptime=10;useinbandfec=1"}
+var redCodecCapability = webrtc.RTPCodecCapability{MimeType: sfu.MimeTypeAudioRed, ClockRate: 48000, Channels: 2, SDPFmtpLine: "111/111"}
```
Comment (boks1971) [inline pkg/rtc/mediaengine.go:13]: Is 111 going to work always? Is this dependent on what client uses for opus payload? Or a pure Pion dependency?

**Review by boks1971 — APPROVED** [review]
Awesome stuff @cnderrauber .

One thing I was thinking is not doing checks in 'forwardRTP' function. Can we start different goroutines in SFU.Receiver based on receiver type to avoid check in the per packet path.

**Review by davidzhao — APPROVED** [review]

File: pkg/rtc/mediaengine.go (hunk 1 above)
Comment (cnderrauber) [reply pkg/rtc/mediaengine.go:13]: For publisher, in `ParticipantImpl.setCodecPreferencesOpusRedForPublisher`, will use the real opus payload
```
		if strings.EqualFold(codec.Name, "red") && strings.Contains(codec.Fmtp, strconv.FormatInt(int64(opusPayload), 10)) {
			preferredCodecs = append(preferredCodecs, strconv.FormatInt(int64(codec.PayloadType), 10))
		} else {
			leftCodecs = append(leftCodecs, strconv.FormatInt(int64(codec.PayloadType), 10))
		}
```
For subscriber, the answer side should use server's payload type, so it is safe to use that.

**Comment by cnderrauber** [comment]
> Awesome stuff @cnderrauber .
> 
> One thing I was thinking is not doing checks in 'forwardRTP' function. Can we start different goroutines in SFU.Receiver based on receiver type to avoid check in the per packet path.

Agree the type check and cast looks not good after we have more than one auxiliary receivers. But if we forwarding it at different goroutines, the `forwardRTP` still need to wait other forward goroutines complete packet spread to make forward to all downtracks synchronize. Maybe callback function is a way, will try that.

**Pull request merged**